src/Tribe/Attendees.php

```diff
@@ -820,6 +820,32 @@ public function maybe_generate_csv( $all = false ) {
 		exit;
 	}
 
+	/**
+	 * Determines if the current user is allowed to export Attendees list as a CSV.
+	 *
+	 * @since TBD
+	 *
+	 * @return boolean
+	 */
+	public function user_can_export_attendees_csv() {
+		$user = wp_get_current_user();
+
+		$allowed_roles = [ 'administrator', 'editor' ];
+
+		$can_export = ! empty( array_intersect( $allowed_roles, $user->roles ) );
```

Comment:
I don't like checking roles instead of capabilities to be honest. Can we check an admin's cap and an editor's cap instead ?

Checking for roles, we are forcing the role to be given to a user that needs to perform that action (along with all the other caps that come with the role). Just checking a cap, we allow for users to create custom roles with a set of caps, or specifically grant users (or roles) a cap instead.

Reply:
https://wp-kama.com/note/wp-capabilities-list we could use a cap admin and editors share only (and of course super admins)

Reply:
@dpanta94 I agree. I don't like checking for roles either and that's why I had `current_user_can( 'manage_options' )` previously there. But,[ Jack's request](https://lw.slack.com/archives/C0714HQSMEW/p1727293533685739?thread_ts=1727206613.079549&cid=C0714HQSMEW) was to add a filter to allow for custom roles.

Reply:
Jack mention adding a filter for the roles, not just the `$can_export` value. If we're going to change to checking roles, then add a filter for that, instead.

Reply:
Otherwise, I would do `$can_export = current_user_can( 'manage_options' )` and then use the filter you have at the bottom.

Reply:
Yes that makes @codingmusician . I made the change, thank you both.